llvm-19.yaml

```diff
@@ -47,75 +65,473 @@ pipeline:
+            ${{targets.contextdir}}/usr/share/bash-completion/completions/clang
+
+          mkdir -p ${{targets.contextdir}}/${{vars.llvm-prefix}}/bin
+          mv ${{targets.destdir}}/${{vars.llvm-prefix}}/bin/*clang* ${{targets.contextdir}}/${{vars.llvm-prefix}}/bin/
+          mv ${{targets.destdir}}/${{vars.llvm-prefix}}/bin/amdgpu-arch ${{targets.contextdir}}/${{vars.llvm-prefix}}/bin/
+          mv ${{targets.destdir}}/${{vars.llvm-prefix}}/bin/nvptx-arch ${{targets.contextdir}}/${{vars.llvm-prefix}}/bin/
+
+          mkdir -p ${{targets.contextdir}}/usr/bin
+          for path in ${{targets.contextdir}}/${{vars.llvm-prefix}}/bin/*; do
+            name=${path##*/}
+            ln -s ../lib/${{package.name}}/bin/$name ${{targets.contextdir}}/usr/bin/$name
+          done
+
+          mkdir -p ${{targets.contextdir}}/${{vars.llvm-prefix}}/lib
+          mv ${{targets.destdir}}/${{vars.llvm-prefix}}/lib/*clang* ${{targets.contextdir}}/${{vars.llvm-prefix}}/lib/
+          mv ${{targets.destdir}}/${{vars.llvm-prefix}}/lib/libfindAllSymbols.a ${{targets.contextdir}}/${{vars.llvm-prefix}}/lib/
+
+          mkdir -p ${{targets.contextdir}}/${{vars.llvm-prefix}}/lib/cmake
+          mv ${{targets.destdir}}/${{vars.llvm-prefix}}/lib/cmake/*clang* ${{targets.contextdir}}/${{vars.llvm-prefix}}/lib/cmake/
+
+          mkdir -p ${{targets.contextdir}}/${{vars.llvm-prefix}}/share
+          mv ${{targets.destdir}}/${{vars.llvm-prefix}}/share/*clang* ${{targets.contextdir}}/${{vars.llvm-prefix}}/share/
+
+          mkdir -p ${{targets.contextdir}}/${{vars.llvm-prefix}}/include
+          mv ${{targets.destdir}}/${{vars.llvm-prefix}}/include/*clang* ${{targets.contextdir}}/${{vars.llvm-prefix}}/include/
+    test:
+      pipeline:
+        - uses: test/compiler-hardening-check
+          with:
+            cc: clang-${{vars.major-version}}
+          runs: |
+            amdgpu-arch --version
+            amdgpu-arch --help
+            clang --version
+            clang --help
+            clang++ --version
+            clang++ --help
+            clang-${{vars.major-version}} --version
+            clang-${{vars.major-version}} --help
+            clang-cl --version
+            clang-cpp --version
+            clang-cpp --help
+            clang-include-cleaner --version
+            clang-include-cleaner --help
+            clang-linker-wrapper --version
+            clang-linker-wrapper --help
+            clang-nvlink-wrapper --help
+            clang-pseudo --version
+            clang-pseudo --help
+            nvptx-arch --version
+            nvptx-arch --help
+
+  - name: llvm-cmake-${{vars.major-version}}
+    description: CMake macros for LLVM ${{vars.major-version}}
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.contextdir}}/${{vars.llvm-prefix}}/share/cmake
+          mv ./cmake/* ${{targets.contextdir}}/${{vars.llvm-prefix}}/share/cmake/
+
```

Comment:
i thought these need to be generated and are templated and need to be installed, not just copied. or are they generated in the current working directory? looks interesting.

Reply:
I'll double check that

Reply:
with         -DCMAKE_INSTALL_PREFIX=${{vars.llvm-prefix}} \ -DCMAKE_PREFIX_PATH=${{vars.llvm-prefix}} now added i expect this package to not exists. cmake files to be shipped in the top level dev package. And it can have provides llvm-cmake-${{vars.major-version}} for compat. But we really don't need this subpackage.

I guess worth rebuilding whatever was using this - rust?!

Reply:
There isn't a package that explicitly depends on llvm-cmake-19 so I've omitted in the latest push. The only package I found that has an explicit dependency on LLVMs CMake headers is qdrant, but it was pinned to an older version so I've bumped it to use the latest Clang